### PR TITLE
Restructure reference

### PR DESCRIPTION
**Pull request by Grifs**

### CODE REVIEW

File: _src/automation/generate_config_schema_export_pages.py [hunk 1]
```diff
@@ -1,329 +1,115 @@
...
-def generate_pages():
-	"""Feed JsonEntryData lists to page generation functions."""
-	# print('\n'.join([ str(r) for r in functionality_entry_list]))
-	# print('\n'.join([ str(r) for r in requirements_entry_list]))
-	generate_combined_page(entry_list= functionality_entry_list, page_title= "Functionality", save_filename= "functionality", title_is_name=False)
-	generate_combined_page(entry_list= requirements_entry_list, page_title= "Setup Requirements", save_filename= "requirements", title_is_name=True)
-	generate_grouped_pages(entry_list= platforms_entry_list, save_dir= config_dir + "/platforms")
-	generate_grouped_pages(entry_list= arguments_entry_list, save_dir= config_dir + "/arguments")
+	# Fix description markdown keywords to links
+	for d in sorted_dict:
```
Comment (DriesSchaumont) [inline _src/automation/generate_config_schema_export_pages.py:56]:
```python3
try:
    d['description'] = replace_keywords(d['description'])
except KeyError:
    pass
```

File: _src/automation/generate_config_schema_export_pages.py [hunk 2]
```diff
@@ -1,329 +1,115 @@
...
-				qmd += qmd_parse_type(data.type)
-				qmd += qmd_removed_deprecated(data.removed, data.deprecated)
-				qmd += qmd_paragraph(replace_keywords(data.description))
-				qmd += qmd_parse_examples(data.examples)
-			else:
-				qmd += qmd_paragraph(replace_keywords(data.description))
-				qmd += qmd_parse_examples(data.examples)
-
-	write_qmd_file(directory=config_dir, file_name=save_filename, content=qmd)
+	if filename in config_pages_settings['structure']:
```
Comment (DriesSchaumont) [inline _src/automation/generate_config_schema_export_pages.py:67]:
```python3
try:
    filename = config_pages_settings['structure'][filename]
except KeyError:
    print(f"Could not find {filename} in the config pages settings structure")
```

File: _src/automation/generate_config_schema_export_pages.py [hunk 3]
```diff
@@ -1,329 +1,115 @@
...
 
-		# Find keyword in csv
-		keywords_csv = open(keyword_replace_csv)
-		csvreader = csv.reader(keywords_csv)
-		for row in csvreader:
-			if row[0] == keyword:
-				link = row[1]
-				break
-		keywords_csv.close()
+		if keyword in config_pages_settings['keywords']:
```
Comment (DriesSchaumont) [inline _src/automation/generate_config_schema_export_pages.py:101]: Could also use try-except here

File: _src/automation/generate_cli_schema_export_pages.py [hunk 4]
```diff
@@ -42,7 +46,7 @@ def create_page(name, json_entry):
 		for subcommand_json in json_entry["subcommands"]:
```
Comment (DriesSchaumont) [inline _src/automation/generate_cli_schema_export_pages.py:46]:
```python3
qmd = "".join([qmd_add_command(subcommand_jon, is_subcommand=True) for subcommand_jon in json
_entry["subcommands"]])
```

File: _src/automation/generate_cli_schema_export_pages.py [hunk 5]
```diff
@@ -130,14 +134,8 @@ def replace_keywords(text: str) -> str:
...
 
-		# Find keyword in csv
-		keywords_csv = open(keyword_replace_csv)
-		csvreader = csv.reader(keywords_csv)
-		for row in csvreader:
-			if row[0] == keyword:
-				link = row[1]
-				break
-		keywords_csv.close()
+		if keyword in config_pages_settings['keywords']:
```
Comment (DriesSchaumont) [inline _src/automation/generate_cli_schema_export_pages.py:137]:
```python3
try:
   link = config_pages_settings['keywords'][keyword]
except KeyError:
   link = "no-link"

File: _src/automation/generate_config_schema_export_pages.py [hunk 6]
```diff
@@ -1,329 +1,115 @@
...
-	if data.topic == TOPIC_FUNCTIONALITY:
-		functionality_entry_list.append(data)
-	elif data.topic == TOPIC_ARGUMENTS:
-		arguments_entry_list.append(data)	
-	elif data.topic == TOPIC_PLATFORMS:
-		platforms_entry_list.append(data)	
-	elif data.topic == TOPIC_REQUIREMENTS:
-		requirements_entry_list.append(data)
+	# Sort data entries alphabetically on 'name'
+	sorted_dict = sorted(json_data, key=lambda x: x["name"], reverse=False)
```
Comment (DriesSchaumont) [inline _src/automation/generate_config_schema_export_pages.py:53]: The sorting of dicts may or may not work depending on the python version
I would turn this into a list of tupes (key, object) if you want to make sure it is sorted.
You can iterate over that with:

```python3

sorted_json_entries = sorted([(item_key, json_obj) for item_key, json_ob in json_data.items()], key=lambda x: x["name"])

for (key, value) in sorted_json_entries:
    ...
```

**Review by DriesSchaumont — CHANGES_REQUESTED** [review]
Small suggestions here and there, being a bit pedantic on some of them. Looks good overall!

File: _src/automation/generate_config_schema_export_pages.py [hunk 7]
```diff
@@ -1,329 +1,115 @@
...
-	generate_grouped_pages(entry_list= arguments_entry_list, save_dir= config_dir + "/arguments")
+	# Fix description markdown keywords to links
+	for d in sorted_dict:
+		if 'description' in d:
+			d['description'] = replace_keywords(d["description"])
 
-def generate_combined_page(entry_list, page_title, save_filename, title_is_name):
-	"""
-	Parses a list of JsonEntryData and combines it into a single page.
+	page_data = {}
```
[the remark this answers is not in the document]
Comment (Grifs) [reply _src/automation/generate_config_schema_export_pages.py:60]: Good point!

File: _src/automation/generate_cli_schema_export_pages.py (hunk 4 above)
Comment (Grifs) [reply _src/automation/generate_cli_schema_export_pages.py:46]: This script still needs to be updated to our new system of using jinja templates, so this most likely will be changed. But if not, I'll keep it in mind

File: _src/automation/generate_config_schema_export_pages.py (hunk 6 above)
Comment (Grifs) [reply _src/automation/generate_config_schema_export_pages.py:53]: Downside of changing inherited code; I just realized the 'sorted_dict' was a misnomer at best. It's actually a list!

**Comment by rcannood** [comment]
The output looks good! 

Comments related to the content itself, not the components:

* I think `Nextflow Legacy Platform` should be under Platforms, not under Nextflow Vdsl3 Platform
* I would expect [/reference/config/platforms/docker/setup/](https://viash.io/pr-preview/pr-27/reference/config/platforms/docker/setup/) to contain a larger example and pointers to the different kinds of requirements, but atm it seems to be a combination of all of its child pages. Is this doable?
* I would have Functionality and Platforms show one more level by default. Atm it's too much clicking.
* Rendering of the table of `viash ns exec` fails: [https://viash.io/pr-preview/pr-27/reference/cli/ns.html#viash-ns-exec](https://viash.io/pr-preview/pr-27/reference/cli/ns.html#viash-ns-exec)
* Nextflow Vdsl3 platform should have a child 'Nextflow Directives'.

**Comment by Grifs** [comment]
> I would expect [/reference/config/platforms/docker/setup/](https://viash.io/pr-preview/pr-27/reference/config/platforms/docker/setup/) to contain a larger example and pointers to the different kinds of requirements, but atm it seems to be a combination of all of its child pages. Is this doable?

I removed the page for now. This is an old version of the setup documentation. This is currently not re-generated either.

**Review by rcannood — APPROVED** [review]
LGTM! Can you move the unchecked item into a separate issue? :bow:

**Comment by Grifs** [comment]
> Can you move the unchecked item into a separate issue? bow

I implemented it instead (please review)

**Comment by rcannood** [comment]
This branch is continued in #31

**Pull request closed without merging**